packages/payment-processor/src/payment/utils.ts

```diff
@@ -264,3 +264,27 @@ export function getAmountToPay(
   }
   return amountToPay;
 }
+
+/**
+ * Pads an amount to match Chainlink's own currency decimals (eg. for fiat amounts).
+ */
+export const padAmountForChainlink = (amount: BigNumberish, currency: Currency) => {
+  let decimalPadding: number;
+  switch (currency.type) {
+    case RequestLogicTypes.CURRENCY.ISO4217: {
+      const chainlinkFiatDecimal = 8;
+      decimalPadding = Math.max(chainlinkFiatDecimal - currency.getDecimals(), 0);
+      break;
+    }
+    case RequestLogicTypes.CURRENCY.ERC20: {
+      decimalPadding = 0;
+      break;
+    }
+    default:
+      throw new Error(
+        'Unsupported request currency for conversion with Chainlink. The request currency has to be fiat or ERC20.',
```

Comment:
Maybe give a CTA like "contact us for missing currency" ?

Reply:
Currencies are constraints by Request Invoicing, users won't see these errors.